### DISCUSSION TITLE
Confused about the code while true in the find bar.

### DISCUSSION THREAD

**Discussion by OriLiMu** [ScratchAddons/ScratchAddons#7606] [Help] [answered]
The code is [here](https://github.com/ScratchAddons/ScratchAddons/blob/master/addons/find-bar/userscript.js#L858)

I cant see the meaning of the `while true` here. I remove it the `find-bar` still works fine.

```js
 while (true) {
    const root = await addon.tab.waitForElement("ul[class*=gui_tab-list_]", {
      markAsSeen: true,
      reduxEvents: ["scratch-gui/mode/SET_PLAYER", "fontsLoaded/SET_FONTS_LOADED", "scratch-gui/locales/SELECT_LOCALE"],
      reduxCondition: (state) => !state.scratchGui.mode.isPlayerOnly,
    });
    findBar.createDom(root);
  }
```

**Comment by mxmou** [comment, marked as answer]
The loop is necessary to show the find bar UI again after switching to the project page and back to the editor. Many addons contain similar code.
